plugin.py

```diff
@@ -45,6 +59,12 @@ def on_workspace_configuration(self, params, configuration) -> None:
                 if workspace_folder:
                     configuration['workspaceFolder'] = workspace_folder.to_lsp()
                 self.resolve_working_directory(configuration, scope_uri, workspace_folder)
+                buf = session.get_session_buffer_for_uri_async(scope_uri)
+                if buf:
+                    configuration['validate'] = self.compute_validate(buf.language_id, scope_uri, configuration)
+                else:
+                    configuration['validate'] = 'on'
+                del configuration['probe']
```

Comment:
Why is this `del` needed?

Reply:
It's not needed but also the server doesn't need this setting. I've done it to save a few bytes during transfer.
I'm fine with removing this line if you insist. :)
I know deleting properties usually doesn't play nice with compilers and internal optimizations but at the same time this is not a hot path.

Reply:
I don't mind either way, was just confused about it :)